microSALT/utils/referencer.py

```diff
@@ -134,92 +140,101 @@ def fetch_external(self, force=False):
+                        st_link = entry.find("./mlst/database/profiles/url").text
+                        service: str = get_service_by_url(st_link)
+                        if service == "pasteur":
+                            database: str = f"pubmlst_{organ.split('_')[0]}_seqdef"
+                            self.set_client(service, database=database)
+                        else:
+                            self.set_client(service)
+
+                        # Parse the database name and scheme ID
+                        try:
+                            parsed_data = self.client.parse_url(url=st_link)
+                        except InvalidURLError as e:
+                            self.logger.warning(f"Invalid URL: {st_link} - {e}")
+                            continue
+
+                        scheme_id = parsed_data.get("scheme_id")  # Extract scheme ID
+                        db = parsed_data.get("db")  # Extract database name
+
+                        if not db or not scheme_id:
+                            self.logger.warning(
+                                f"Could not extract database name or scheme ID from MLST URL: {st_link}"
+                            )
+                            return
+
+                        scheme_info = self.client.retrieve_scheme_info(
+                            db, scheme_id
+                        )  # Retrieve scheme info
+                        last_updated = scheme_info.get("last_updated")  # Extract last updated date
+                        if (
+                            int(last_updated.replace("-", "")) <= int(currver.replace("-", ""))
+                            and not force
+                        ):
+                            self.logger.info(
+                                f"Profile for {organ.replace('_', ' ').capitalize()} already at the latest version."
+                            )
+                            continue
                         self.logger.info(
-                            f"Profile for {organ.replace('_', ' ').capitalize()} already at the latest version."
+                            f"pubMLST reference for {organ.replace('_', ' ').capitalize()} updated to {last_updated} from {currver}"
                         )
-                        continue
-                    self.logger.info(
-                        f"pubMLST reference for {organ.replace('_', ' ').capitalize()} updated to {last_updated} from {currver}"
-                    )
 
-                    # Step 1: Download the profiles CSV
-                    st_target = f"{self.config['folders']['profiles']}/{organ}"
-                    profiles_csv = self.client.download_profiles_csv(db, scheme_id)
+                        # Step 1: Download the profiles CSV
+                        st_target = f"{self.config['folders']['profiles']}/{organ}"
+                        profiles_csv = self.client.download_profiles_csv(db, scheme_id)
```

Comment:
Would recommend using the `csv` package in the python standard library, to get a more robust parsing:
```python
import csv
with open("file.csv") as csvfile:
    csvrd = csv.reader(csvfile)
    for row in csvrd:
        # etc etc
```

Reply:
I don't disagree with you, but it seems out of scope, I have not really changed this logic in this PR, only added authentication to the process